core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/ImagePathInDiagramCheck.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2022 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.platform.sirius.sirius.validation.ddiagram;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.log4j.Logger;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.validation.IValidationContext;
+import org.eclipse.emf.validation.model.ConstraintStatus;
+import org.eclipse.sirius.business.api.dialect.DialectManager;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.diagram.WorkspaceImage;
+import org.eclipse.sirius.diagram.tools.internal.validation.constraints.ImagePathConstraint;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.polarsys.capella.common.tools.report.config.registry.ReportManagerRegistry;
+import org.polarsys.capella.common.tools.report.util.IReportManagerDefaultComponents;
+import org.polarsys.capella.core.data.capellacore.CapellaElement;
+import org.polarsys.capella.core.model.handler.validation.CapellaDiagnostician;
+
+/**
+ * Check if the images used on diagram elements, can be found.</br>
+ * Warning: This rule will load all representation associated to the validation context(Capella element)
+ * 
+ * @author lfasani
+ */
+public class ImagePathInDiagramCheck extends ImagePathConstraint {
+
+  protected Logger logger = ReportManagerRegistry.getInstance().subscribe(IReportManagerDefaultComponents.VALIDATION);
+
+  protected StringBuilder desc = null;
+
+  @Override
+  public IStatus validate(final IValidationContext ctx) {
+    // Ugly code to check only if called by the Capella Validation
+    // Indeed, otherwise, this code is called many times by Sirius and it is too impacting to change Sirius.
+    boolean calledByCapella = Arrays.asList(Thread.getAllStackTraces().get(Thread.currentThread())).stream()
```

Comment:
Can you replace `Arrays.asList(Thread.getAllStackTraces().get(Thread.currentThread())).stream()` by 
`Stream.of(Thread.currentThread().getStackTrace())`

Reply:
done